libraries/platforms/dromajo-vcs/mcbin.S

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2020, University of Washington All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without modification,
+# are permitted provided that the following conditions are met:
+#
+# Redistributions of source code must retain the above copyright notice, this list
+# of conditions and the following disclaimer.
+#
+# Redistributions in binary form must reproduce the above copyright notice, this
+# list of conditions and the following disclaimer in the documentation and/or
+# other materials provided with the distribution.
+#
+# Neither the name of the copyright holder nor the names of its contributors may
+# be used to endorse or promote products derived from this software without
+# specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+# ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# This assembly file includes the manycore binary if there is one
+
+.section .rodata.manycore
+
+.globl manycore
+.align 4
+.type manycore, @object
+manycore:
+.incbin BSG_MANYCORE_KERNELS
+
+.globl manycore_size
+.type manycore_size, @object
+.align 4
+manycore_size:
+.int manycore_size - manycore
```

Comment:
This file is used to generate another assembly file where the BSG_MANYCORE_KERNELS flow variable is replaced with its value here during compilation. One caveat with running SPMD and CUDA tests is that you have to run the `*.log` target directly (i.e. you can't run `main.exec` and then run `main.exe.log`). This has to do with the visibility of the BSG_MANYCORE_KERNELS variable in the flow.